matrix-sdk/src/main/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandler.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.androidsdk.data.room;
+
+import android.content.Context;
+import android.text.TextUtils;
+
+import org.matrix.androidsdk.R;
+import org.matrix.androidsdk.data.Room;
+import org.matrix.androidsdk.data.RoomState;
+import org.matrix.androidsdk.rest.model.RoomMember;
+import org.matrix.androidsdk.util.Log;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * This class helps to compute a room display name
+ */
+public class RoomDisplayNameHandler {
+
+    private static final String LOG_TAG = RoomDisplayNameHandler.class.getSimpleName();
+
+    private final Room mRoom;
+
+    public RoomDisplayNameHandler(Room room) {
+        mRoom = room;
+    }
+
+    public String handle(Context context) {
+        try {
+            // this algorithm is the one defined in
+            // https://github.com/matrix-org/matrix-js-sdk/blob/develop/lib/models/room.js#L617
+            // calculateRoomName(room, userId)
+
+            // For Lazy Loaded room, see algorithm here:
+            // https://docs.google.com/document/d/11i14UI1cUz-OJ0knD5BFu7fmT6Fo327zvMYqfSAR7xs/edit#heading=h.qif6pkqyjgzn
+
+            RoomState roomState = mRoom.getState();
+
+            if (!TextUtils.isEmpty(roomState.name)) {
+                return roomState.name;
+            }
+
+            if (!TextUtils.isEmpty(roomState.getCanonicalAlias())) {
+                return roomState.getCanonicalAlias();
+            }
+
+            // Temporary patch: use the first alias if available
+            if (roomState.aliases != null && !roomState.aliases.isEmpty()) {
+                return roomState.aliases.get(0);
+            }
+
+            List<RoomMember> othersActiveMembers = new ArrayList<>();
+            List<RoomMember> activeMembers = new ArrayList<>();
```

Comment:
`activeMembers` array is not really useful now
You should remove it, and just look for the currentUserMember (RoomMember)